Operator Overloading in C++ (#2315)
girlscript/winter-of-contributing: C_CPP/Operator Overloading/operator_overloading_in_c++.md

}

AddArray::AddArray(int num)
{
sizeArr = num;
cout << "Enter the elements of the array :\n";
for (int i = 0; i < sizeArr; ++i)
{
cin >> arr[i]; //input array elements
}
}

void AddArray::display()
{
cout << "Displaying the elements of the array :\n";
for (int i = 0; i < sizeArr; ++i)
{
cout << arr[i] << " ";
}
cout << "\n";
}

AddArray AddArray::operator+(AddArray &arr2) //member function to overload + operator
{
AddArray obj;
obj.sizeArr = sizeArr;
for (int i = 0; i < sizeArr; ++i)
{
obj.arr[i] = arr[i] + arr2.arr[i]; //adding elements of the two arrays
}
return obj; //returning the object of AddArray
}

int main()
{
int num;
cout << "Enter the size of the array : ";
cin >> num;
cout << "\n";

AddArray obj1(num), obj2(num), obj3;
obj3 = obj1 + obj2; //invokes + operator by implicitly issuing
                    //the member function call obj1.operator+(obj2)
cout << "\nThe resultant array is :\n";
obj3.display();

return 0;

}
```

output:

```c
Enter the size of the array : 5

Enter the elements of the array :
67 54 13 32 90
Enter the elements of the array :
23 45 65 88 97

The resultant array is :
Displaying the elements of the array :        
90 99 78 120 187
```

> 
> ```c
> **Code for operator overloading implemented using non-member function**
> 
> /*Add two arrays using operator overloading implemented through non-member (friend) function*/
> 
> #include <iostream>

Reviewer: ```suggestion

**Code for operator overloading implemented using non-member function**

```c
/*Add two arrays using operator overloading implemented through non-member (friend) function*/

#include <iostream>
```

using namespace std;

class AddArray
{
private:
    int arr[5];
    int sizeArr;

> public:
>     AddArray();                     //defaut constructor
>     AddArray(int);                  //paremeterized constructor
>     void display();                 //function for displaying the elements of an array
>     friend AddArray operator+(AddArray ,AddArray ); //non-member function to overload + operator
> };
> 
> AddArray::AddArray()
> {
>     for (int i = 0; i < 5; ++i)
>     {

Reviewer: ```suggestion
public:
    AddArray();                     //default constructor
    AddArray(int);                  //parameterized constructor
    void display();                 //function for displaying the elements of an array
    friend AddArray operator+(AddArray ,AddArray ); //non-member function to overload +operator
};

AddArray::AddArray()
{
    for (int i = 0; i < 5; ++i)
    {
```

#Operator Overloading

> 

Reviewer: ```suggestion
# Operator Overloading

```

add this space
